### PR TITLE
Allow build on FreeBSD (and clang)

### PR DESCRIPTION
**Pull request by apparentorder**
Some minor changes to allow building on FreeBSD.

Tested on FreeBSD 12.0R on a RPi2 (arm).

### CODE REVIEW

File: sdr.c [hunk 1]
```diff
@@ -113,10 +113,11 @@ void sdrShowHelp()
 
 bool sdrHandleOption(int argc, char **argv, int *jptr)
 {
+    int i = 0;
```
Comment (mutability) [inline sdr.c]: This seems like a C dialect issue? C99/C11 shouldn't have trouble with this.

**Comment by mutability** [comment]
Hey! Sorry about the huge delay on getting back to this. Are there any updates needed here for more recent freebsd before I merge?

**Comment by apparentorder** [comment]
> Hey! Sorry about the huge delay on getting back to this. Are there any updates needed here for more recent freebsd before I merge?

Actually, with all the changes to the Makefile since the original PR, almost no changes are required. I've adjusted the PR commits accordingly.

Ad hoc I'm not sure what RTLSDR_PREFIX would do, but as long as it's not used, `-lusb*` isn't needed at all.

It compiles and runs on FreeBSD 12.1R (arm / Pi) this way.

**Comment by mutability** [comment]
Current dev branch builds OK for me on FreeBSD 12.1-RELEASE (I updated the README with some notes) and started in net-only mode; real SDRs not tested. Can you try it out?

**Comment by apparentorder** [comment]
> Current dev branch builds OK for me on FreeBSD 12.1-RELEASE (I updated the README with some notes) and started in net-only mode; real SDRs not tested. Can you try it out?

Tested successfully!

Just one minor thing about the README: Why would you build pkgconf from the ports and all other dependencies from packages? This command just work just fine: `pkg install rtl-sdr bladerf hackrf pkgconf gmake`.

**Comment by mutability** [comment]
Hm, I think I went looking for pkgconf via `pkg` and didn't find it, hence ports. Let me have another poke at that VM.

File: README.md [hunk 2]
```diff
@@ -56,6 +56,13 @@ package-wheezy/. Build in there (`dpkg-buildpackage -b`)
 
 The wheezy build does not include bladeRF, HackRF, or LimeSDR support.
 
+## Building on FreeBSD
+
+To build, you will need librtlsdr, pkgconf and GNU Make. For users of pkg(8):
+`pkg install rtl-sdr pkgconf gmake`.
```
Comment (bovine) [inline README.md:62]: Should this line be?
``` suggestion
`pkg install rtl-sdr bladerf hackrf pkgconf gmake`.
```

Comment (apparentorder) [reply README.md:62]: My understanding is that we'll use the README.md section that's already been committed to the 'dev' branch in e00d12f. At least that's the version I was asking @mutability about yesterday.

Comment (mutability) [reply README.md:62]: Yep, the changes in this PR were merged as part of #38

**Comment by mutability** [comment]
I retested on the freebsd VM and there is. indeed, no reason why we can't use `pkg` for pkgconf here; can't remember for sure what the problem I hit the first time around was, but I think I may have initially been looking for `pkg-config` not `pkgconf`. I've updated the README.

With that, I think everything here is merged, closing.

**Pull request closed without merging**